sys/kern/dev_procstat.c

```diff
@@ -0,0 +1,125 @@
+#include <sys/devfs.h>
+#include <sys/vnode.h>
+#include <sys/linker_set.h>
+#include <sys/queue.h>
+#include <sys/proc.h>
+#include <sys/thread.h>
+#include <sys/klog.h>
+#include <sys/malloc.h>
+#include <stdio.h>
+
+/* Implementation of /dev/procstat
+ *
+ * When /dev/procstat is opened it takes a snapshot of info of all existing
+ * processes.
+ *
+ * During read calls to /dev/procstat this info can be read.
+ *
+ * Example:
+ * euid   pid    ppid    pgrp    state    td_name  elfpath
+ * 0       1       0       1       R       init    /bin/ksh
+ *
+ */
+
+#define PSTAT_SIZE 1024
+#define BUF_SIZE (PSTAT_SIZE - sizeof(int))
```

Comment:
`pstat_t::len` is `size_t` so if I understand correctly you should use `size_t` here instated of `int`.

Reply:
yes, thanks